sound/soc/sof/loader.c

```diff
@@ -864,5 +864,15 @@ void snd_sof_fw_unload(struct snd_sof_dev *sdev)
 	/* TODO: support module unloading at runtime */
 	release_firmware(sdev->pdata->fw);
 	sdev->pdata->fw = NULL;
+
+	/*
+	 * TODO: Currently we are not able to wake DSP up once it is suspeded.
+	 * Therefore pm_runtime_get_sync() is called by sof_ipc4_init_msg_memory()
+	 * to prevent DSP from entering suspend. Call pm_runtime_put_noidle()
+	 * here to balance it.
+	 */
+	if (snd_sof_get_abi_major(sdev) >= 4)
```

Comment:
It worth the effort to compile test things locally _before_ pushing changes ;)

Push only if you are certain that what you have is what you want.